src/main.rs

```diff
@@ -10,33 +9,17 @@ use tokio::sync::mpsc;
 struct Cli {
     pattern: String,
 
-    #[clap(parse(from_os_str), multiple_values = true)]
-    paths: Vec<std::path::PathBuf>,
-}
-
-fn convert_paths_to_utf8(paths: Vec<std::path::PathBuf>) -> Vec<Utf8PathBuf> {
-    let mut results: Vec<Utf8PathBuf> = Vec::new();
-    let mut errors = false;
-    for path in paths {
-        match Utf8PathBuf::from_path_buf(path) {
-            Ok(value) => results.push(value),
-            _ => errors = true,
-        }
-    }
-    if errors {
-        warn!(
-            "Discarded some file paths because they had ivalid encoding. Valid file paths: {}",
-            results.iter().map(|p| p.to_string()).collect::<Vec<String>>().join(", ")
-        );
-    }
-    results
+    #[clap(multiple_values = true)]
+    paths: Vec<PathBuf>,
```

Comment:
you're better at defensiveness with the `convert_paths_to_utf8`, but then we don't really need `camino` and could stick with the regular `std::path::PathBuf` (and deal with non UTF-8 paths). My suggestion for `camino` was because in most cases they will be UTF-8 or a subset, so it simplifies the code in general.